src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppNameGenerator.java

```diff
@@ -72,14 +73,21 @@ public void afterPropertiesSet() throws Exception {
 	@Override
 	public String generateAppName(String appName) {
 		if (StringUtils.isEmpty(prefixToUse)) {
-			return appName;
+			return appName.substring(0, Math.min(MAX_APPNAME_LENGTH, appName.length()));
 		} else {
-			return String.format("%s-%s", prefixToUse, appName);
+			String string = String.format("%s-%s", prefixToUse, appName);
+			return string.substring(0, Math.min(MAX_APPNAME_LENGTH, string.length()));
 		}
 	}
 
 	private String createUniquePrefix() {
```

Comment:
Because I realized it's really not needed. We're not really hashing anything even